hikari/events/channel_events.py

```diff
@@ -780,3 +788,264 @@ async def fetch_guild_webhooks(self) -> typing.Sequence[webhooks.PartialWebhook]
+    thread_id: snowflakes.Snowflake = attr.field()
+    # <<inherited docstring from GuildThreadEvent>>.
+
+    guild_id: snowflakes.Snowflake = attr.field()
+    # <<inherited docstring from GuildThreadEvent>>.
+
+    parent_id: snowflakes.Snowflake = attr.field()
+    """The ID of the channel that the thread was deleted from."""
+
+    type: channels.ChannelType = attr.field()
+    """The type of thread that was deleted."""
+
+
+@base_events.requires_intents(intents.Intents.GUILDS)
+@attr_extensions.with_copy
+@attr.define(kw_only=True, weakref_slot=False)
+class OwnThreadMemberUpdateEvent(GuildThreadEvent):
+    """Event fired when the bot's own member is updated in a thread."""
+
+    app: traits.RESTAware = attr.field(metadata={attr_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from Event>>.
+
+    shard: gateway_shard.GatewayShard = attr.field(metadata={attr_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from ShardEvent>>.
+
+    member: channels.ThreadMember = attr.field()
+    """The member that was updated."""
+
+    guild_id: snowflakes.Snowflake = attr.field()
+    # <<inherited docstring from GuildThreadEvent>>.
+
+    @property
+    def thread_id(self) -> snowflakes.Snowflake:
+        # <<inherited docstring from GuildThreadEvent>>.
+        return self.member.thread_id
+
+
+@base_events.requires_intents(intents.Intents.GUILDS | intents.Intents.GUILD_MEMBERS)
+@attr_extensions.with_copy
+@attr.define(kw_only=True, weakref_slot=False)
+class ThreadMembersUpdateEvent(GuildThreadEvent):
+    """Event fired when other thread members are updated."""
+
+    app: traits.RESTAware = attr.field(metadata={attr_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from Event>>.
+
+    shard: gateway_shard.GatewayShard = attr.field(metadata={attr_extensions.SKIP_DEEP_COPY: True})
+    # <<inherited docstring from ShardEvent>>.
+
+    thread_id: snowflakes.Snowflake = attr.field()
+    # <<inherited docstring from GuildThreadEvent>>.
+
+    guild_id: snowflakes.Snowflake = attr.field()
+    # <<inherited docstring from GuildThreadEvent>>.
+
+    approximate_member_count: int = attr.field(eq=False, hash=False, repr=True)
+    """Approximate count of members in the thread channel.
+
+    !!! warning
+        This stops counting at 50.
+    """
+
+    added_members: typing.Mapping[snowflakes.Snowflake, channels.ThreadMember] = attr.field()
+    """Mapping of IDs to objects of the members that were added to the thread."""
+
+    removed_member_ids: typing.Sequence[snowflakes.Snowflake] = attr.field()
+    """Sequence of IDs of users that were removed from the thread."""
+
+    guild_members: typing.Mapping[snowflakes.Snowflake, guilds.Member] = attr.field()
+    """Mapping of IDs to guild member objects of the added thread members."""
+
+    guild_presences: typing.Mapping[snowflakes.Snowflake, presences.MemberPresence] = attr.field()
+    """Mapping of IDs to guild presence objects of the added members."""
```

Comment:
Does this require presences intent to be enabled?

Reply:
yes